### DISCUSSION TITLE
Confusion about the Tooltip hooks/component

### DISCUSSION THREAD

**Discussion by i-am-the-slime** [airbnb/visx#1285] [General]
Hello there, this is a great library.

I'm wondering if the tooltip brings any data-vis related advantages or if it's more "agnostic" and can be used all over a project.

**Comment by williaster** [comment]
Hey @i-am-the-slime 👋  thanks for checking out `visx`. Sorry for the delayed reply. I'm going to move this to a discussion since it's a usage question and not an issue.

Overall you can find the docs for `@visx/tooltip` [here](https://airbnb.io/visx/docs/tooltip). It provides two very different components:
- `useTooltip` and the older-style `withTooltip` higher-order component: which simply manage tooltip **state** and have nothing to do with visually rendering a tooltip, e.g., whether it's open/closed, its data, its top/left position, and functions to update that state
- `<ToolTip />`, `<TooltipWithBounds />`, and `<TooltipInPortal />`: these are actually visible components and theoretically could be used in any app. They are essentially tooltip containers and can be used in conjunction with `useTooltip` state; they accept `top/left` positions as well as `children` tooltip content. The bounds detection and rendering in a portal could theoretically be used in any app as well, tho they solve problems common in data vis engineering. 

I'd definitely check out the [`tooltip` demo](https://airbnb.io/visx/tooltip) if you haven't, to get a sense for what they do/their usage. That demo actually has nothing to do with charts. 

Hope that helps!
